packages/kbn-performance-testing-dataset-extractor/src/es_client.ts

```diff
@@ -172,9 +177,45 @@ export class ESClient {
     return await this.getTransactions<TransactionDocument>(queryFilters);
   }
 
-  async getSpans(transactionId: string) {
-    const filters = [{ field: 'parent.id', value: transactionId }];
-    const queryFilters = filters.map((filter) => addBooleanFilter(filter));
-    return await this.getTransactions<SpanDocument>(queryFilters);
+  getMsearchRequestItem = (queryFilters: QueryDslQueryContainer[]): MsearchRequestItem => {
+    return {
+      query: {
+        bool: {
+          filter: [
+            {
+              bool: {
+                filter: queryFilters,
+              },
+            },
+          ],
+        },
+      },
+    };
+  };
+
+  async getSpans(transactionIds: string[]) {
+    const searches = new Array<MsearchRequestItem>();
+
+    for (const transactionId of transactionIds) {
+      const filters = [{ field: 'parent.id', value: transactionId }];
+      const queryFilters = filters.map((filter) => addBooleanFilter(filter));
+      const requestItem = this.getMsearchRequestItem(queryFilters);
+      searches.push({}, requestItem);
+    }
+    this.log.debug(`Msearch request: ${JSON.stringify(searches)}`);
+    const result = await this.client.msearch<SpanDocument>({
+      searches,
+    });
```

Comment:
I feel like we should be filtering by index at least, so that this doesn't run the query on every index but only where necessary